util/args.py

```diff
@@ -208,24 +179,16 @@ def get_args() -> argparse.Namespace:
         help="Number of ProtoTrees to train and (optionally) use in an ensemble. Used in main_ensemble.py",
     )
     args = parser.parse_args()
-    args.milestones = get_milestones(args)
+    args.milestones_list = get_milestones_list(args.milestones)  # TODO Seems a bit hacky to put this in args.
     return args
 
 
-"""
-    Parse the milestones argument to get a list
-    :param args: The arguments given
+def get_milestones_list(milestones_str: str):
     """
-
-
-def get_milestones(args: argparse.Namespace):
-    if args.milestones != "":
-        milestones_list = args.milestones.split(",")
-        for m in range(len(milestones_list)):
-            milestones_list[m] = int(milestones_list[m])
-    else:
-        milestones_list = []
-    return milestones_list
+        Parse the milestones argument to get a list
+        :param milestones_str: The milestones as a comma separated string, e.g. "23,34,45"
+    """
+    return list(map(int, milestones_str.split(","))) if milestones_str else []
 
 
 def save_args(args: argparse.Namespace, directory_path: Union[str, Path]) -> None:

```

Comment:
all saving related code can probably be deleted. We will let pytorch lightning take care of this. If you want, that can be done in a followup PR instead of here

Reply:
Makes sense, I've now deleted this code in this PR.